Models/global_config.hpp

```diff
@@ -0,0 +1,51 @@
+#include <cstdint>
+
+typedef enum {
+    yaw,
+    pitch,
+    roll,
+    thrust
+} ControlAxis_t;
+
+typedef enum {
+    positive,
+    negative
+} ControlDirection_t;
+
+//TODO: Determine correct location for motor driver definitions, and finish implementation.
+class MotorDriver{};
+class TempPWMDriver : public MotorDriver{};
+class TempDSHOTDriver : public MotorDriver{};
+
+typedef struct {
+    ControlAxis_t axis;
+    ControlDirection_t direction;
+    MotorDriver driver;
+    //TODO: Consider a more specific datatype for interfaceID
+    //(e.g. a datatype defined within the motor driver class)
+    uint8_t interfaceID;
+} Motor_t;
+
+typedef struct {
+    float p;
+    float i;
+    float d;
+} PID_t;
+
+typedef struct {
+    float min;
+    float max;
+} AxisLimit_t;
+
+typedef struct {
+    AxisLimit_t yawLimit;
+    AxisLimit_t pitchLimit;
+    AxisLimit_t rollLimit;
+    AxisLimit_t thrustLimit;
+} ControlLimits_t;
+
+//TODO: Combine flightmode classes with AM control algorithms
+class Flightmode{
```

Comment:
This is a class with no public members

Reply:
I do understand that it's a placeholder, but It may be more appropriate to just rename the `ControlAlgorithm` class and include it here?

Reply:
Yes I plan to something along those lines when integrating with AM.